src/pkg/cluster/injector.go

```diff
@@ -115,59 +113,82 @@ func (c *Cluster) StartInjectionMadness(ctx context.Context, tmpDir string, imag
 			GracePeriodSeconds: &deleteGracePeriod,
 			PropagationPolicy:  &deletePolicy,
 		}
-		err := c.Clientset.CoreV1().Pods(ZarfNamespaceName).Delete(ctx, "injector", deleteOpts)
+		selector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"app": "zarf-injector",
+			},
+		})
+		if err != nil {
+			return err
+		}
+		listOpts := metav1.ListOptions{
+			LabelSelector: selector.String(),
+		}
+		err = c.Clientset.CoreV1().Pods(ZarfNamespaceName).DeleteCollection(ctx, deleteOpts, listOpts)
 		if err != nil {
-			message.Debug("could not delete pod injector:", err)
+			return err
 		}
 
-		// Update the podspec image path and use the first node found
-
 		pod, err := c.buildInjectionPod(node[0], image, payloadConfigmaps, sha256sum)
 		if err != nil {
-			// Just debug log the output because failures just result in trying the next image
-			message.Debug("error making injection pod:", err)
-			continue
+			return fmt.Errorf("error making injection pod: %w", err)
 		}
 
-		// Create the pod in the cluster
 		pod, err = c.Clientset.CoreV1().Pods(pod.Namespace).Create(ctx, pod, metav1.CreateOptions{})
 		if err != nil {
-			// Just debug log the output because failures just result in trying the next image
-			message.Debug("error creating pod in cluster:", pod, err)
-			continue
+			return fmt.Errorf("error creating pod in cluster: %w", err)
 		}
 
-		// if no error, try and wait for a seed image to be present, return if successful
-		if c.injectorIsReady(ctx, seedImages, spinner) {
-			spinner.Success()
-			return
+		objs := []object.ObjMetadata{
+			{
+				GroupKind: schema.GroupKind{
+					Kind: "Pod",
+				},
+				Namespace: ZarfNamespaceName,
+				Name:      pod.Name,
+			},
 		}
-
+		waitCtx, waitCancel := context.WithTimeout(ctx, 60*time.Second)
+		defer waitCancel()
+		err = pkgkubernetes.WaitForReady(waitCtx, c.Watcher, objs)
+		if err != nil {
+			return err
+		}
+		spinner.Success()
 		// Otherwise just continue to try next image
 	}
-
-	// All images were exhausted and still no happiness
-	spinner.Fatalf(nil, "Unable to perform the injection")
```

Comment:
We should return an error here, before the program was entirely stopping as spinner.Fatalf calls os.exit(1)

Reply:
The `spinner.Fatalf` was necessary before because we were logging errors in the for loop instead of returning them, so it used to be possible for the loop to complete even when errors occurred, but that is no longer possible because we return any errors we encounter immediately, which means if any iteration of the loop successfully creates the pod, the function will immediately return nil. The error path will never reach this branch if the function encounters any error. That was acting as a catch-all for errors

Reply:
Actually we should probably only loop to get an image and after that do everything outside of a for loop. Since we switched the continue statements to returns. The only thing that worries me a bit is I don't have a good sense for how flaky the injector process is so I don't know if those continue statements were actually getting used

Reply:
In that case we should only loop over the images until we have an acceptable image, then run everything else outside of the loop. 

The only worry I have is that I don't have a good sense if the continue blocks were getting used. I'd lean towards saying it's a good simplification though

Reply:
The loop over images may not be necessary. We plan to simplify this logic in #2621. Since the scope of this PR has already expanded, let’s handle that refactor separately

Reply:
There was some confusion cleared up in convo, basically just simplified the code a bit. We will still need to do https://github.com/defenseunicorns/zarf/issues/2621 separately